hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodieClientMultiWriter.java

```diff
@@ -531,34 +531,35 @@ public void testMultiWriterWithAsyncTableServicesWithConflict(HoodieTableType ta
 
         // Since the concurrent modifications went in, this upsert has
         // to fail
+        // Note: the start time of upsert must be smaller than the compaction scheduling time
```

Comment:
hi, why this upsert commit must be smaller than compaction scheduling time, can you describe the problem in detail.